pkg/analyzer/analyzers/notion/notion.go

```diff
@@ -0,0 +1,389 @@
+	req, err := http.NewRequest(h.Method, h.Endpoint, data)
+	if err != nil {
+		return false, err
+	}
+
+	// Add custom headers if provided
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	// Execute HTTP Request
+	resp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	// Check response status code
+	switch {
+	case StatusContains(resp.StatusCode, h.ValidStatuses):
+		return true, nil
+	case StatusContains(resp.StatusCode, h.InvalidStatuses):
+		return false, nil
+	default:
+		return false, errors.New("error checking response status code")
+	}
+}
+
+type Scope struct {
+	Name     string         `json:"name"`
+	HttpTest HttpStatusTest `json:"test"`
+}
+
+func readInScopes() ([]Scope, error) {
+	var scopes []Scope
+	if err := json.Unmarshal(scopesConfig, &scopes); err != nil {
+		return nil, err
+	}
+
+	return scopes, nil
+}
+
+func getPermissions(cfg *config.Config, key string) ([]string, error) {
+	scopes, err := readInScopes()
+	if err != nil {
+		return nil, fmt.Errorf("reading in scopes: %w", err)
+	}
+
+	permissions := make([]string, 0, len(scopes))
+	for _, scope := range scopes {
+		status, err := scope.HttpTest.RunTest(cfg, map[string]string{"Authorization": "Bearer " + key, "Notion-Version": "2022-06-28"})
+		if err != nil {
+			return nil, fmt.Errorf("running test: %w", err)
+		}
+		if status {
+			permissions = append(permissions, scope.Name)
+		}
+	}
+
+	return permissions, nil
+}
+
+type SecretInfo struct {
+	Bot            *bot
+	WorkspaceUsers []user
+	Permissions    []string
+}
+
+type user struct {
+	Id     string `json:"id"`
+	Name   string `json:"name"`
+	Type   string `json:"type"`
+	Person struct {
+		Email string `json:"email"`
+	}
+}
+
+type bot struct {
+	Id   string `json:"id"`
+	Name string `json:"name"`
+	Type string `json:"type"`
+	Bot  struct {
+		Owner *struct {
+			Type string `json:"type"`
+		}
+		WorkspaceName string `json:"workspace_name"`
+	} `json:"bot"`
+}
+
+func (b *bot) GetWorkspaceName() string {
+	return b.Bot.WorkspaceName
+}
+
+func (b *bot) OwnedBy() string {
+	if b.Bot.Owner != nil {
+		return b.Bot.Owner.Type
+	}
+	return "N/A"
+}
+
+func AnalyzeAndPrintPermissions(cfg *config.Config, key string) {
+	info, err := AnalyzePermissions(cfg, key)
+	if err != nil {
+		color.Red("[x] Error : %s", err.Error())
+		return
+	}
+
+	color.Green("[!] Valid Notion API key\n\n")
+
+	color.Green("[i] Bot: %s (%s)\n", info.Bot.Name, info.Bot.Id)
+	color.Green("[i] Bot Owned By: %s\n", info.Bot.OwnedBy())
+
+	if info.Bot.GetWorkspaceName() != "" {
+		color.Green("[i] Workspace: %s\n\n", info.Bot.GetWorkspaceName())
+	}
+
+	printPermissions(info.Permissions)
+	if len(info.WorkspaceUsers) > 0 {
+		printUsers(info.WorkspaceUsers)
+	}
+	color.Yellow("\n[i] Expires: Never")
+
+}
+
+func AnalyzePermissions(cfg *config.Config, key string) (*SecretInfo, error) {
+	permissions := make([]string, 0)
+
+	bot, err := getBotInfo(cfg, key)
```

Comment:
suggestion: It looks like we create a client in both `getBotInfo` and `getWorkspaceUsers` using `analyzers.NewAnalyzeClient(cfg)`. Could we instantiate the client once and reuse it in both functions? While the impact might be minimal here, reusing connections could be beneficial if many analyzers are running simultaneously.